yabause/src/yglshaderes.c

```diff
@@ -958,6 +958,52 @@ static YglVdp1CommonParam grow_tess = { 0 };
 static YglVdp1CommonParam grow_spd_tess = { 0 };
 static YglVdp1CommonParam mesh_tess = { 0 };
 
+/*------------------------------------------------------------------------------------
+*  VDP1 Half luminance Operaion
+* ----------------------------------------------------------------------------------*/
+const GLchar Yglprg_vdp1_half_luminance_v[] =
+#if defined(_OGLES3_)
+      "#version 300 es \n"
+#else
+      "#version 330 \n"
+#endif
+      "uniform mat4 u_mvpMatrix;    \n"
+    "uniform vec2 u_texsize;    \n"
+      "layout (location = 0) in vec4 a_position;   \n"
+      "layout (location = 1) in vec4 a_texcoord;   \n"
+      "out   vec4 v_texcoord;     \n"
+      "void main()                  \n"
+      "{                            \n"
+      "   gl_Position = a_position*u_mvpMatrix; \n"
+      "   v_texcoord  = a_texcoord; \n"
+    "   v_texcoord.x  = v_texcoord.x / u_texsize.x;\n"
+    "   v_texcoord.y  = v_texcoord.y / u_texsize.y;\n"
+      "} ";
+const GLchar * pYglprg_vdp1_half_luminance_v[] = {Yglprg_vdp1_half_luminance_v, NULL};
+
+const GLchar Yglprg_vpd1_half_luminance_f[] =
+#if defined(_OGLES3_)
+      "#version 300 es \n"
+#else
+      "#version 330 \n"
+#endif
+      "precision highp float;                            \n"
+      "in vec4 v_texcoord;                            \n"
+      "uniform sampler2D s_texture;                        \n"
+      "out vec4 fragColor;            \n"
+      "void main()                                         \n"
+      "{                                                   \n"
+      "  vec2 addr = v_texcoord.st;                        \n"
+      "  addr.s = addr.s / (v_texcoord.q);                 \n"
+      "  addr.t = addr.t / (v_texcoord.q);                 \n"
+      "  vec4 FragColor = texture( s_texture, addr );      \n"
+      "  /*if( FragColor.a == 0.0 ) discard;*/                \n"
+      "  fragColor = FragColor;\n "
+      "  fragColor.a = 0.5;\n "

```

Comment:
Half luminance is not half transparent.

I think it should be like this. Right?
```C
fragColor.r = FragColor.r /2.0;
fragColor.g = FragColor.g /2.0;
fragColor.b = FragColor.b /2.0;
fragColor.a = FragColor.a;
```